File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
class Solution {
    public int solution(String t, String p) {
        int answer = 0;
        int pLength = p.length();
        long pNum = Long.parseLong(p);  // p를 숫자로 변환
        
        // t에서 p와 길이가 같은 부분 문자열을 추출하여 비교
        for (int i = 0; i <= t.length() - pLength; i++) {
            String subStr = t.substring(i, i + pLength);  // 부분 문자열 추출
            long subNum = Long.parseLong(subStr);  // 부분 문자열을 숫자로 변환
            
            if (subNum <= pNum) {  // p보다 작거나 같으면
                answer++;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.py
def solution(t, p):
    answer = 0
    len_p = len(p)
    int_p = int(p)
    
    for i in range(len(t) - len_p + 1):
        sub = t[i:i+len_p]
        if int(sub) <= int_p:
            answer += 1
            
    return answer
